packages/common/src/store/pages/profile/reducer.ts

```diff
@@ -43,86 +43,169 @@ const initialState = {
 }
 
+const initialState = { currentUser: null, entries: {} }
+
 const actionsMap = {
   [FETCH_PROFILE](state, action) {
-    if (action.fetchOnly) return state
+    const { fetchOnly, shouldSetLoading, handle, userId } = action
+    if (fetchOnly) return state
+
+    const profileState = state.entries[handle] ?? initialProfileState
 
     const newState = {
-      ...state,
-      status: action.shouldSetLoading ? Status.LOADING : state.status
+      ...profileState,
+      status: shouldSetLoading ? Status.LOADING : state.status
     }
-    if (action.handle) {
-      newState.handle = action.handle
+    if (handle) {
+      newState.handle = handle
     }
-    if (action.userId) {
-      newState.userId = action.userId
+    if (userId) {
+      newState.userId = userId
+    }
+    return {
+      currentUser: handle,
+      entries: { ...state.entries, [handle]: newState }
     }
-    return newState
   },
   [FETCH_PROFILE_SUCCEEDED](state, action) {
-    if (action.fetchOnly) return state
+    const { currentUser, entries } = state
+    const { fetchOnly, userId, handle } = action
+    const profileHandle = handle ?? currentUser
+    if (fetchOnly) return state
+
+    const profileState = entries[profileHandle]
 
     return {
       ...state,
-      status: Status.SUCCESS,
-      userId: action.userId,
-      handle: action.handle
+      entries: {
+        ...entries,
+        [profileHandle]: {
+          ...profileState,
+          status: Status.SUCCESS,
+          userId,
+          handle: profileHandle
+        }
+      }
     }
   },
   [FETCH_FOLLOW_USERS](state, action) {
+    const { currentUser, entries } = state
+    const { followerGroup, handle } = action
+    const profileHandle = handle ?? currentUser
+    const profileState = entries[profileHandle]
+
     return {
       ...state,
-      [action.followerGroup]: {
-        ...state[action.followerGroup],
-        status: Status.LOADING
+      entries: {
+        ...entries,
+        [profileHandle]: {
+          ...profileState,
+          [followerGroup]: {
+            ...profileState[followerGroup],
+            status: Status.LOADING
+          }
+        }
       }
     }
   },
   [FETCH_FOLLOW_USERS_SUCCEEDED](state, action) {
-    const filteredAddedUserIds = action.userIds.filter(({ id }) =>
-      state[action.followerGroup].userIds.every(
-        ({ id: userId }) => id !== userId
-      )
+    const { currentUser, entries } = state
+    const { userIds, followerGroup, handle } = action
+    const profileHandle = handle ?? currentUser
+    const filteredAddedUserIds = userIds.filter(({ id }) =>
+      state[followerGroup].userIds.every(({ id: userId }) => id !== userId)
     )
+
+    const profileState = entries[profileHandle]
+
     return {
       ...state,
-      [action.followerGroup]: {
-        userIds:
-          state[action.followerGroup].userIds.concat(filteredAddedUserIds),
-        status: Status.SUCCESS
+      entries: {
```

Comment:
Starting to look like we should convert this to redux toolkit so that we can leverage immer

Reply:
yeah def agreed